### PR TITLE
cluster: ngm server default enabled during deployment

### LINKED ISSUE #1690: tiup cluster ：ngm server default enabled during deployment
## Feature Request

**Is your feature request related to a problem? Please describe:**

**Describe the feature you'd like:**

For new deployed v5.4.0+ tidb cluster or any old version cluster which is upgraded to 5.4.0 + version , tiup will enable ngm server during the deployment or upgrade. 

special case: 
--  If the user sets  ng_port = 0 in config, we will not deploy ngm server. 
--  If the user uses old version tiup cluster  components which does not contain this new feature, tiup will not default enable ngm server.

**Why the featue is needed:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by nexustar**
### What problem does this PR solve? 

- close #1690 
- also fix data_dir is wrong when upgrade CDC on some version

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

Release notes:

```release-note
NONE
```

### CODE REVIEW

File: pkg/cluster/spec/validate.go [hunk 1]
```diff
@@ -569,8 +569,8 @@ func (s *Specification) portInvalidDetect() error {
 			if strings.HasSuffix(compSpec.Type().Field(i).Name, "Port") {
 				port := int(compSpec.Field(i).Int())
 				portTags := strings.Split(compSpec.Type().Field(i).Tag.Get("yaml"), ",")
-				// when use not specify ng_port, its default value is 0
-				if port == 0 && len(portTags) > 1 && portTags[1] == "omitempty" {
+				// when port has omitempty tag, it means it is a optional port whice 0 means default and -1 means diable
+				if len(portTags) > 1 && portTags[1] == "omitempty" {
 					continue
 				}
 				if port < 1 || port > 65535 {
```
Comment (AstroProfundis) [inline pkg/cluster/spec/validate.go:576]: do we need to allow `-1` here?

Comment (nexustar) [reply pkg/cluster/spec/validate.go:576]: yep, allow any int value for port with omitempty tag

File: pkg/cluster/spec/validate.go [hunk 2]
```diff
@@ -568,9 +568,8 @@ func (s *Specification) portInvalidDetect() error {
 		for i := 0; i < compSpec.NumField(); i++ {
 			if strings.HasSuffix(compSpec.Type().Field(i).Name, "Port") {
 				port := int(compSpec.Field(i).Int())
-				portTags := strings.Split(compSpec.Type().Field(i).Tag.Get("yaml"), ",")
-				// when use not specify ng_port, its default value is 0
-				if port == 0 && len(portTags) > 1 && portTags[1] == "omitempty" {
+				// for NgPort, 0 means default and -1 means diable
```
Comment (AstroProfundis) [inline pkg/cluster/spec/validate.go:571]:
```suggestion
				// for NgPort, 0 means default and -1 means disable
```

**Review by AstroProfundis — APPROVED** [review]

**Pull request merged**